HW4

BoLjshOJ opened the pull request:
HW4 done

BoLjshOJ committed:
init HW4

BoLjshOJ committed:
HW 4 Перевести приложение для проведения опросов на Spring Shell
-Подключить Spring Shell, используя starter

stvort commented on src/main/java/ru/otus/springFramework/shell/Starter.java:
```diff
@@ -0,0 +1,21 @@
+
+    @ShellMethod("Start Quiz")
+    public void start(){
+        quizService.startTest();
```
Давайте все же попробуем сделать хотя бы 2 команды login и старт, чтобы попробовать обеспечить порядок их вызова с помощью ```@ShellMethodAvailability```

BoLjshOJ replied:
resolved by 951ff3958884b8a114589b941e1d97bfe73906ce

BoLjshOJ committed:
HW4
1.Давайте все же попробуем сделать хотя бы 2 команды login и старт, чтобы попробовать обеспечить порядок их вызова с помощью @ShellMethodAvailability
-add 2 command in shell
-update "start" method with Availability

stvort commented on src/main/java/ru/otus/springFramework/shell/Starter.java:
```diff
@@ -0,0 +1,40 @@
+    private final QuizService quizService;
+    private final StudentService studentService;
+    private Student newStudent;
+    private boolean login;
```
Лучше как-то типа isLoggedIn

BoLjshOJ replied:
resolved by 11f775338365d99cff6800d58064fa8b6712d810

BoLjshOJ committed:
HW4
fix last comments

BoLjshOJ committed:
Merge branch 'master' into HW4

# Conflicts:
#	pom.xml
#	src/main/java/ru/otus/springFramework/Application.java
#	src/main/java/ru/otus/springFramework/dao/QuestionDaoImpl.java
#	src/main/java/ru/otus/springFramework/service/EntityPrinter.java
#	src/main/java/ru/otus/springFramework/service/QuizService.java
#	src/main/java/ru/otus/springFramework/service/QuizServiceImpl.java
#	src/test/java/ru/otus/springFramework/dao/StudentDaoSimpleTest.java
